Sources/web3swift/Web3/Web3+InfuraProviders.swift

```diff
@@ -4,316 +4,101 @@
 //  Copyright © 2018 Alex Vlasov. All rights reserved.
 //
 import Foundation
-import BigInt
 import Starscream
 
-public enum BlockNumber {
-    case pending
-    case latest
-    case earliest
-    case exact(BigUInt)
-
-    public var stringValue: String {
-        switch self {
-        case .pending:
-            return "pending"
-        case .latest:
-            return "latest"
-        case .earliest:
-            return "earliest"
-        case .exact(let number):
-            return String(number, radix: 16).addHexPrefix()
-        }
-    }
-}
```

Comment:
I've just started to reviewing this PR, thinking it could take a while (like up to the end of this week).

But just yet i have such question like what the purpose of this dropping `BlockNumber` enum?

I see that you've making an abstraction over the API spec, which is more close to given enum that to any literal type itself.